### PR TITLE
Extract shared example for cacheable response in specs

### PR DESCRIPTION
**Pull request by mjankowski**

### CODE REVIEW

File: spec/support/examples/cache.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    expect(response.headers['Set-Cookies']).to be_nil
+  end
+
+  it 'does not set sessions' do
+    expect(session).to be_empty
+  end
+
+  if expects_vary
+    it 'returns Vary header' do
+      expect(response.headers['Vary']).to include 'Accept, Accept-Language, Cookie'
```
Comment (ClearlyClaire) [inline spec/support/examples/cache.rb:15]: I would prefer if the value was not hardcoded but came from `expects_vary`.

**Review by ClearlyClaire — COMMENTED** [review]
This is an improvement, but it probably only applies to controller specs (while we really want that kind of check to be done in requests specs). Is there a way we can make that more explicit?

**Comment by mjankowski** [comment]
> This is an improvement, but it probably only applies to controller specs (while we really want that kind of check to be done in requests specs). Is there a way we can make that more explicit?

Is what you mean there that your first choice would be that all of the examples checking for caching behavior are done in request specs, instead of a mix of controller and request?

**Comment by ClearlyClaire** [comment]
Ideally, examples checking for caching behavior should be done in request specs, yes. Short term, the kind of refactoring this PR does is fine, but it should be clear that the shared examples are about controllers, and that we should move away from it.

File: spec/support/examples/cache.rb (hunk 1 above)
Comment (mjankowski) [reply spec/support/examples/cache.rb:15]: As in ... the cases which DO need a `Vary` value here should explicitly pass in a string with the values they expect to find? (or false when they dont?)

Comment (ClearlyClaire) [reply spec/support/examples/cache.rb:15]: Sorry, I realize that I haven't replied to this. Yes, that's what I meant, I suggest `expects_vary` to be either false or a string, and in the latter case have it be used for comparison. Even if that does add a little duplication, I think it's better if it's explicit here.

Comment (mjankowski) [reply spec/support/examples/cache.rb]: Rebased, and pushed another commit which flips the default to false and requires a string be passed in (which gets checked if present).

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**